### PR TITLE
refactor(tools): use fs.promises API

### PR DESCRIPTION
**Pull request by saschanaz**
Uses `fs.promises` which is [now marked as stable](https://nodejs.org/api/fs.html#fs_fs_promises_api).

### CODE REVIEW

File: tools/builder.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 "use strict";
 const colors = require("colors");
-const fsp = require("fs-extra");
```
Comment (sidvishnoi) [inline tools/builder.js]: We can remove `fs-extra` from package.json now, right?

**Comment by sidvishnoi** [comment]
Should we update this too?
https://github.com/w3c/respec/blob/2c073773d343e3a93196ad20307d5508a3df83d0/package.json#L7

File: tools/builder.js (hunk 1 above)
Comment (saschanaz) [reply tools/builder.js]: `copydeps` still uses `fs-extra` for its own utility functions, but maybe we could cook our own ones.

**Comment by saschanaz** [comment]
>Should we update this too?

Ah yes... but should we defer this until version 8 losses its support? Not sure.

**Comment by sidvishnoi** [comment]
`fs.promises` isn't available in node 8, right? It'll break if use node 11 features there

File: tools/builder.js (hunk 1 above)
Comment (sidvishnoi) [reply tools/builder.js]: I checked. `cp` is being used only for files (not directories), so we can use the native one easily. And we have `fs.rmdir` and `fs.unlink` for `fsp.remove` 🎉

**Comment by saschanaz** [comment]
The API is introduced in Node.js 10, but yes, it will break 8.

**Comment by saschanaz** [comment]
>This is in node lts, right?

Sure 😃

**Comment by marcoscaceres** [comment]
We should target active lts only. Ok to delete 8.

**Comment by saschanaz** [comment]
One notable thing is that https://github.com/nodejs/node/pull/26581 is not backported into 10 LTS yet, so using ReSpec there may emit some warnings.

**Comment by saschanaz** [comment]
Like this... Looks like we should wait a bit more 😅

image

**Comment by sidvishnoi** [comment]
Lets pipe it through `grep -v "ExperimentalWarning: The fs.promises API is experimental"` 😛

**Comment by marcoscaceres** [comment]
The warnings are ok IMO. Reading the history, the API did not change from when it was experimental to when it became stable... so technically, it was stable. Also, the warning don’t affect processing, so I think it’s fine to ship.

File: tools/builder.js (hunk 1 above)
Comment (sidvishnoi) [reply tools/builder.js]: @saschanaz Do you plan to fix above here or separate PR?

Comment (saschanaz) [reply tools/builder.js]: Hmm, maybe a separate one.

**Pull request merged**